src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiAdapterFactoryProcessor.java

```diff
@@ -131,6 +163,47 @@ private ExecutableElement getJsonAdapterMethod(Element element) {
     return null;
   }
 
+  private void error(Element element, String message, Object... args) {
+    if (args.length > 0) {
+      message = String.format(message, args);
+    }
+    processingEnv.getMessager().printMessage(ERROR, message, element);
+  }
+
+  private boolean containsJsonAdapterFactoryMirror(List<? extends TypeMirror> interfaces) {
```

Comment:
Good point, wouldn't hurt to support it. Would something like [this](https://github.com/rharter/auto-value-parcel/blob/master/auto-value-parcel/src/main/java/com/ryanharter/auto/value/parcel/TypeSimplifier.java#L11) work or should I do the full walk up the ancestry?